divi/src/ActiveChainManager.cpp

```diff
@@ -103,30 +96,29 @@ bool ActiveChainManager::DisconnectBlock(
         const TransactionLocationReference txLocationReference(tx, pindex->nHeight, transactionIndex);
         const auto* undo = (transactionIndex > 0 ? &blockUndo.vtxundo[transactionIndex - 1] : nullptr);
         const TxReversalStatus status = UpdateCoinsReversingTransaction(tx, txLocationReference, view, undo);
-        if(!CheckTxReversalStatus(status,fClean))
-        {
-            return false;
-        }
-        if(!pfClean)
-        {
+        if(!fJustCheck)
             IndexDatabaseUpdateCollector::ReverseTransaction(tx,txLocationReference,view,indexDBUpdates);
-        }
+
+
+        const bool oldClean = fClean;
+        const bool reversalOk = CheckTxReversalStatus(status, fClean);
+        /* Print an error message for the first transaction that fails (at least).  */
+        if (!reversalOk || (oldClean && !fClean))
+            error("%s: error reverting transaction %s in block %s at height %d",
+                  __func__, tx.GetHash(), block.GetHash(), pindex->nHeight);
+        if (!reversalOk)
+            return false;
```

Comment:
Here you can just `return error(...);` since the error method returns false always after logging.

Reply:
This would change the semantics, though, as we only want to early return on `!reversalOk`, but want to log an error about the first transaction that is not clean.  (But see also the other comment.)